packages/web/src/components/userprofile/EditUserForm.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import {
+  Button,
+  ButtonGroup,
+  FormControl,
+  FormLabel,
+  FormHelperText,
+  Input,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalBody,
+  ModalCloseButton,
+  useDisclosure,
+} from '@chakra-ui/react';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+import { User } from '../../pages/user/[uid]';
+
+type EditFormValues = yup.InferType<typeof editFormSchema>;
+const editFormSchema = yup.object({
+  name: yup.string().trim().required('Name is required.'),
+  pronouns: yup.string(),
+  schoolName: yup.string(),
+});
+
+export interface EditUserProps {
+  setEditToggle(editToggle: boolean): void;
+  setUser(user: User): void;
+  user: {
+    id: string;
+    name: string;
+    pronouns?: string;
+    schoolName?: string;
+  };
+}
+
+export const EditUserForm: React.FC<EditUserProps> = (props: EditUserProps) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [isModalShown, setIsModalShown] = React.useState<boolean>(false);
+
+  const formik = useFormik<EditFormValues>({
+    initialValues: {
+      name: props.user.name,
+      pronouns: props.user.pronouns,
+      schoolName: props.user.schoolName,
+    },
+    validationSchema: editFormSchema,
+    onSubmit: async (data) => {
+      const res = await fetch(`/api/users/${props.user.id}`, {
+        method: 'PATCH',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(data),
+      });
+      if (res.status === 200) {
+        props.setUser(await res.json());
+        props.setEditToggle(false);
+      } else {
```

Comment:
Remember to always add a `try catch` around things that can throw! The fetch and the `res.json` can both throw.

Also `res.ok` should always be used instead of checking the status code if you're simply trying to evaluate whether it "worked" (99% of the time)
```suggestion
      const res = await fetch(`/api/users/${props.user.id}`, {
        method: 'PATCH',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(data),
      });
      if (res.status.ok) {
        props.setUser(await res.json());
        props.setEditToggle(false);
      } else {
```